### PR TITLE
Addiing gh action for spelling errors

### LINKED ISSUE #1080: [Feature] GitHub Action to catch spelling errors
## Problem 
Throughout my experience reviewing the sources contributors want to add to LinksHub, there has been something that's been recurring: misspellings in the descriptions section. 
## Possible Solution
Easier one: Creating a workflow based on this [GitHub action](https://github.com/marketplace/actions/check-spelling) <--- This is what I'm leaning towards. 
Possibly hard: Adding this [spell checker](https://www.npmjs.com/package/spelling).

## Additional information
I'm willing to collaborate with any contributor on this issue.

### Screenshots

_No response_

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [x] I am willing to work on this issue (optional)
- [ ] I am a GSSoC'23 contributor

### PR DESCRIPTION
**Pull request by CBID2**
## Fixes Issue
Closes #1080 

## Changes proposed

I added a GitHub Action that encourages contributors to revise spelling errors in their PRs.
## Screenshots

github action

## Note to reviewers
@saikiranreddy201, I'm open to your suggestions on how to improve this. For context, check out the GH Actions link [here](https://github.com/marketplace/actions/check-spelling?version=v0.0.21)

### CODE REVIEW

**Comment by saikiranreddy201** [comment]
Hey @CBID2, Since the "pull_request_target" event already covers "opened," "reopened," and "synchronize" types, there is no need to specify them individually. 

You can achieve this by changing the follow code snippet 
```
pull_request_target:
    branches:
    - "**"
    tags-ignore:
    - "**"
    types:
    - 'opened'
    - 'reopened'
    - 'synchronize'
   ```
   
   to this 
   ```
   pull_request_target:
  branches:
    - "**"
  tags-ignore:
    - "**"
```

**Comment by CBID2** [comment]
> pull_request_target:
> branches:
>  - "**"
> tags-ignore:
>  - "**"

Where should I put it @saikiranreddy201?

File: .github/workflows/spelling.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    - "**"
+  pull_request_target:
+    branches:
+    - "**"
+    tags-ignore:
+    - "**"
+    types:
+    - 'opened'
+    - 'reopened'
+    - 'synchronize'
```
Comment (saikiranreddy201) [inline .github/workflows/spelling.yml:50]: Hey @CBID2, replace this code snippet!

**Comment by saikiranreddy201** [comment]
@CBID2, I have added a comment in the review section, please do checkout!

**Comment by CBID2** [comment]
> Hey @CBID2, Since the "pull_request_target" event already covers "opened," "reopened," and "synchronize" types, there is no need to specify them individually.
> 
> You can achieve this by changing the follow code snippet
> 
> ```
> pull_request_target:
>     branches:
>     - "**"
>     tags-ignore:
>     - "**"
>     types:
>     - 'opened'
>     - 'reopened'
>     - 'synchronize'
> ```
> 
> to this
> 
> ```
> pull_request_target:
> branches:
>  - "**"
> tags-ignore:
>  - "**"
> ```

Thanks

**Comment by ujjawaltyagii** [comment]
@CBID2  The deployment shouldn't have caused error then why deployment is getting failed 0_0

**Comment by CBID2** [comment]
> @CBID2  The deployment shouldn't have caused error then why deployment is getting failed 0_0

I don't know @ujjawaltyagii. Unfortunately, only @rupali-codes has access to Vercel.

**Comment by rupali-codes** [comment]
@CBID2 your vercel deployment is getting failed because of the following issues

image

**Comment by CBID2** [comment]
@k-deepak04, I could use help with this

**Comment by k-deepak04** [comment]
> @k-deepak04, I could use help with this

Didn't get what you have written 
Do you want my help in this?

File: components/Header/Header.tsx [hunk 2]
```diff
@@ -1,3 +1,26 @@
+
+import { FC, useContext } from "react";
```
Comment (pchinjr) [inline components/Header/Header.tsx:2]: These look like duplicate imports, possibly from an incorrect merge

File: components/Header/Header.tsx [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+import Link from "next/link";
+import { AiOutlineMenu } from "react-icons/ai";
+import Logo from "components/logo";
+import { GlobalContext } from "context/GlobalContext";
+import { ThemeToggler } from "../ThemeToggler/themeToggler";
+import { TopBar } from "../TopBar/TopBar";
+import { SocialMediaIconsList } from "components/SocialMedia/SocialMediaIconsList";
+import { useEffect } from "react";
+
+export const Header:FC<{}> = () => {
```
Comment (pchinjr) [inline components/Header/Header.tsx:12]: same here, you declare const Header and it is defined again on line 33

**Comment by CBID2** [comment]
> > @k-deepak04, I could use help with this
> 
> 
> 
> Didn't get what you have written 
> 
> Do you want my help in this? 

Yes

**Comment by k-deepak04** [comment]
> > > @k-deepak04, I could use help with this
> > 
> > 
> > Didn't get what you have written
> > Do you want my help in this?
> 
> Yes

Okay will do, caught up in many things, i will try it our this weekend

**Comment by CBID2** [comment]
> > > > @k-deepak04, I could use help with this
> 
> > > 
> 
> > > 
> 
> > > Didn't get what you have written
> 
> > > Do you want my help in this?
> 
> > 
> 
> > Yes
> 
> 
> 
> Okay will do, caught up in many things, i will try it our this weekend 

Thanks

**Comment by k-deepak04** [comment]
@CBID2 

try using this in header.tsx file

import { FC, useContext, useEffect } from "react";
import Link from "next/link";
import { AiOutlineMenu } from "react-icons/ai";
import Logo from "components/logo";
import { GlobalContext } from "context/GlobalContext";
import { ThemeToggler } from "../ThemeToggler/themeToggler";
import { TopBar } from "../TopBar/TopBar";
import { SocialMediaIconsList } from "components/SocialMedia/SocialMediaIconsList";

interface Props {}

const Header: FC<Props> = () => {
  const { toggleNav } = useContext(GlobalContext);

  useEffect(() => {
    const script = document.createElement('script');
    script.src = 'https://buttons.github.io/buttons.js';
    script.async = true;
    script.defer = true;
    document.head.appendChild(script);
    return () => {
      document.head.removeChild(script);
    };
  }, []);

  return (
    <div className="flex items-center justify-between p-4">
      <div className="flex items-center gap-2">
        <AiOutlineMenu className="text-2xl cursor-pointer" onClick={toggleNav} />
        <Link href="/">
          <a className="font-bold text-xl">
            <Logo />
          </a>
        </Link>
      </div>
      <div className="bg-base-200 relative h-full grow px-8 dark:bg-gray-900 lg:bg-gray-100 lg:dark:bg-[#101623]">
        <Link
          className="github-button"
          href="https://github.com/rupali-codes/LinksHub"
          data-icon="octicon-repo-forked"
          data-size="large"
          data-show-count="true"
          aria-label="Fork rupali-codes/LinksHub on GitHub"
        >
          Fork
        </Link>
        <TopBar className="absolute left-8 hidden h-full md:flex" />
        <div className="absolute right-8 flex h-full gap-4">
          <SocialMediaIconsList className={'hidden lg:flex'} />
        </div>
      </div>
    </div>
  );
};

export default Header;

Before deploying your project on Vercel, make sure to run npm run build or yarn build to generate the optimized build output. 

If you encounter any specific issues or error messages during deployment on Vercel, please provide more details, and I'll be happy to assist you further :)

**Comment by k-deepak04** [comment]
@CBID2 don't know why the return does not have any code in preview , while writing that i was able to see the code.
please let me know how i can share the code.....with you

**Comment by CBID2** [comment]
> @CBID2 don't know why the return does not have any code in preview , while writing that i was able to see the code.
> 
> please let me know how i can share the code.....with you

My preference is [Feedback # 5](https://www.freecodecamp.org/news/code-review-tips/)

**Comment by k-deepak04** [comment]
@CBID2 tried but the deployement is still failing

@rupali-codes please give me the response ss why it's failing

**Comment by rupali-codes** [comment]
@k-deepak04 & @CBID2 its failing because of this

image

**Comment by CBID2** [comment]
> @k-deepak04 & @CBID2 its failing because of this
> 
> 
> 
> image
> 
> 

So what should I do @rupali-codes? I'm not familiar with Typescript

**Comment by k-deepak04** [comment]
> > @k-deepak04 & @CBID2 its failing because of this
> > image
> 
> So what should I do @rupali-codes? I'm not familiar with Typescript

same have experience in nextjs,reactjs not this.

**Comment by rupali-codes** [comment]
@CBID2 & @k-deepak04 there you go!

**Review by CBID2 — COMMENTED** [review]
Done

**Comment by CBID2** [comment]
> @CBID2 & @k-deepak04 there you go!

Thank you @rupali-codes! Now I just need an approved review

**Review by k-deepak04 — APPROVED** [review]

**Review by rupali-codes — COMMENTED** [review]
@CBID2 you have tested this right?

**Comment by CBID2** [comment]
> @CBID2 you have tested this right?

No @rupali-codes. How can I go about doing so?

**Comment by rupali-codes** [comment]
> > @CBID2 you have tested this right?
> 
> No @rupali-codes. How can I go about doing so?

I guess we should merge and lets see :)

**Comment by CBID2** [comment]
> > > @CBID2 you have tested this right?
> 
> > 
> 
> > No @rupali-codes. How can I go about doing so?
> 
> 
> 
> I guess we should merge and lets see :)

Ok, approve my PR

**Review by rupali-codes — APPROVED** [review]

**Comment by k-deepak04** [comment]
image

i have tested one spell checker

**Comment by k-deepak04** [comment]
> image
> 
> i have tested one spell checker

@CBID2 check this, it detects requres spelling and other ones

**Comment by CBID2** [comment]
> > image
> 
> > 
> 
> > i have tested one spell checker
> 
> 
> 
> @CBID2 check this, it detects requres spelling and other ones

Ok

**Pull request merged**